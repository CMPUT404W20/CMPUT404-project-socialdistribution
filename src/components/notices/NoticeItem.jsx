import React, { Component } from "react";
import "../../styles/notices/NoticeItem.scss";
import PropTypes from "prop-types";
import Col from "react-bootstrap/Col";

class NoticeItem extends Component {
  constructor(props) {
    super(props);
    this.props = props;
  }


  render() {
    const {
      username, id, type, handleAccept, handleDecline,
    } = this.props;
    return (
      <Col>
        <div className="notice-item-wrapper">
          {/* need to redirect to the user's profile page */}
          <Col sm={3} md={4}>
            <a href=" " className="username-link">{username}</a>
          </Col>
          <Col sm={3} md={4}>
            <div className="type-wrapper">
              <p>{type}</p>
            </div>
          </Col>
          <Col>
            <div className="button-wrapper">
              <button type="button" className="accept-button" onClick={() => handleAccept(id)}>ACCEPT</button>
              <button type="button" className="decline-button" onClick={() => handleDecline(id)}>DECLINE</button>
            </div>
          </Col>
        </div>

      </Col>

    );
  }
}

NoticeItem.propTypes = {
  username: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  handleAccept: PropTypes.func.isRequired,
  handleDecline: PropTypes.func.isRequired,
};

export default NoticeItem;
