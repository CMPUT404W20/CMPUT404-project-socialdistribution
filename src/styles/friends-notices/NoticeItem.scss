@import "../common.scss";

.wrapper {
    border: 1px solid $border-color;
    margin: 10px;
    padding: 10px 20px 10px 20px;
    border-radius: $radius;
}

.leftColumn {
    height: 50px;

    .username-link {
        margin: 0;
        position: absolute;
        top: 50%;
        -ms-transform: translateY(-50%);
        transform: translateY(-50%);
        text-decoration: none;
        font-weight: 600;
        font-size: 12pt;
        color: black;
    }
}

.middleColumn {
    height: 50px;

    .type-wrapper {
        margin: 0;
        position: absolute;
        top: 50%;
        -ms-transform: translateY(-50%);
        transform: translateY(-50%);
        padding-left: 1rem;
        border-left: 2px solid $accent-color;
    }
}

.rightColumn {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 50px;

    .button-wrapper {
        position: absolute;
        margin-right: 20px;
        right: 0;
    }

    .accept-button, .decline-button {
        @extend %button;
        font-size: 16px;
    }
    
    .decline-button {
        margin: 0 0 0 20px;
        @extend %button;
        background-color: #fa93a7;
        &:not(:disabled) {
            &:hover, &:focus {
                outline: none;
                background-color: #e73a5a;
            }
        }
    }
}