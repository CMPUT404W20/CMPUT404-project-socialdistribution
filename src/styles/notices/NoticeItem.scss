$accent-color: #0275b1;
$type-text-color: rgba(24, 24, 24, 0.6);
.notice-item-wrapper {
    border: 1px solid #BFBFBF;
    box-sizing: border-box;
    padding: 0 1rem 0 2rem;
    border-radius: 4px;
    height: 4rem;
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-bottom: 1rem;
    .username-link {
        text-decoration: none;
        font-weight: 600;
        font-size: 12pt;
        color: black;
        margin-right: 2rem;
        width: 20%;
    }
    .username-link:hover {
        color: $accent-color;
    }
    .type-wrapper {
        display: inline-flex;
        align-items: center;
        padding-left: 2rem;
        height: 2rem;
        width: 30%;
        margin-right: 2rem;
        border-left: 2px solid $accent-color;
        >p {
            margin: 0 0 0 0;
            color: $type-text-color;
        }
    }
    .button-wrapper {
        display: inline-flex;
        flex-direction: row;
    }
    .accept-button,
    .decline-button {
        background-color: $accent-color;
        color: white;
        width: 5rem;
        height: 2rem;
        font-size: 9pt;
        display: inline-flex;
        justify-content: center;
        align-items: center;
        font-style: normal;
        text-align: center;
        -webkit-appearance: none;
        -moz-appearance: none;
        border: none;
        border-radius: 4px;
    }
    .decline-button {
        margin-left: 1rem;
        background-color: white;
        color: #B7B7B7;
        border: 1px solid #EAEAEA;
        box-sizing: border-box;
    }
}