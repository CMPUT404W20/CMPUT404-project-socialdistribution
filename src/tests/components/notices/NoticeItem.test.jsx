import React from "react";
import { shallow } from "enzyme";
import NoticeItem from "../../../components/notices/NoticeItem";

describe("Notice Item Component", () => {
  it("should render correctly", () => {
    const component = shallow(<NoticeItem
      user={{
        id: "", displayName: "", host: "", github: "", url: "",
      }}
      handleDecline={() => {}}
      handleAccept={() => {}}
    />);
    expect(component).toMatchSnapshot();
  });
});
