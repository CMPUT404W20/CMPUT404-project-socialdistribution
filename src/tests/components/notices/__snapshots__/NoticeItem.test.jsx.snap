// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notice Item Component should render correctly 1`] = `
<Row
  className="wrapper"
  noGutters={false}
>
  <Col
    className="leftColumn"
    lg={4}
    md={12}
  >
    <Link
      className="username-link"
      to={
        Object {
          "pathname": "/profile/",
          "state": Object {
            "host": undefined,
            "userID": "",
            "username": "",
          },
        }
      }
    />
  </Col>
  <Col
    className="middleColumn"
    lg={3}
    md={12}
  >
    <ContextConsumer>
      <Component />
    </ContextConsumer>
  </Col>
  <Col
    className="rightColumn"
    lg={5}
    md={12}
  >
    <div
      className="button-wrapper"
    >
      <button
        className="accept-button"
        onClick={[Function]}
        type="button"
      >
        Accept
      </button>
      <button
        className="decline-button"
        onClick={[Function]}
        type="button"
      >
        Decline
      </button>
    </div>
  </Col>
</Row>
`;
